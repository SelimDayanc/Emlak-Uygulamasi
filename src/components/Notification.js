import React from 'react';
import {
  Animated,
  Dimensions,
  Easing,
  StyleSheet,
  Text,
  TouchableHighlight,
  StatusBar,
} from 'react-native';
import colors from './../../common/colors';

export default class Notification extends React.Component {
  static propTypes = {
    message: React.PropTypes.oneOfType([
      React.PropTypes.string,
      React.PropTypes.array,
    ]),
    messageType: React.PropTypes.string,
    actions: React.PropTypes.object.isRequired,
  };

  constructor(props) {
    super(props);
    this.fadeAnim = new Animated.Value(1);
  }

  // componentDidMount() {
  //   StatusBar.setHidden(true);
  // }
  //
  // componentWillUnmount() {
  //   StatusBar.setHidden(false);
  // }

  shouldComponentUpdate(nextProps) {
    return this.props.message !== nextProps.message;
  }

  hideMessage = () => {
    this.props.actions.dismissNotification();
    StatusBar.setHidden(false);
  };

  showMessage = () => {
    StatusBar.setHidden(true);
    Animated.timing(this.fadeAnim, {
      toValue: 1, // Target
      duration: 500, // Configuration
      easing: Easing.bounce,
    }).start();
  };

  componentDidMount() {
    setTimeout(this.hideMessage, 4000);
    this.showMessage();
  }

  render() {
    const {messageType, message} = this.props;

    return (
      <Animated.View
        style={[
          styles.container,
          styles[messageType],
          {opacity: this.fadeAnim},
        ]}>
        <TouchableHighlight onPress={() => this.hideMessage()}>
          <Text style={styles.title}>{message}</Text>
        </TouchableHighlight>
      </Animated.View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
    width: Dimensions.get('window').width,
    backgroundColor: 'transparent',
    height: 64,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 5,
    paddingTop: 20,
    zIndex: 10000,
  },
  title: {
    color: colors.fadedWhite,
    fontSize: 18,
    fontWeight: '100',
    textShadowColor: colors.smokeGreyLight,
    textShadowOffset: {width: 0.2, height: 0.2},
  },
  icon: {
    width: 30,
    height: 30,
    alignSelf: 'center',
    color: colors.accent,
  },
  success: {
    backgroundColor: '#1ecd97',
  },
  error: {
    backgroundColor: '#fb797e',
  },
});
