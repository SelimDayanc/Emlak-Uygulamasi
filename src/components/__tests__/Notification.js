import React from 'react';
import renderer from 'react-test-renderer';
import Notification from '../Notification';

test('renders Notification Component', () => {
  const tree = renderer.create(
    <Notification actions={{}} message="error message" messageType="error" />,
  );
  expect(tree).toMatchSnapshot();
});
