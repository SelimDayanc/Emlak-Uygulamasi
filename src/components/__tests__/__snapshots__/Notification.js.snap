// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Notification Component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fb797e",
      "height": 64,
      "justifyContent": "center",
      "left": 0,
      "opacity": 1,
      "padding": 5,
      "paddingTop": 20,
      "position": "absolute",
      "top": 0,
      "width": 750,
      "zIndex": 10000,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgba(255, 255, 255, .8)",
          "fontSize": 18,
          "fontWeight": "100",
          "textShadowColor": "#e3e7ed",
          "textShadowOffset": Object {
            "height": 0.2,
            "width": 0.2,
          },
        }
      }
    >
      error message
    </Text>
  </View>
</View>
`;
